File: src/cviz/timeline/commands/AmcpCommand.java
package cviz.timeline.commands;

import cviz.ITimeline;
import se.svt.caspar.amcp.AmcpLayer;

import java.util.ArrayList;
import java.util.Arrays;
import java.util.stream.Collectors;
import java.util.stream.Stream;

public class AmcpCommand extends ICommand {
    private String command;

    public AmcpCommand(int layerId, String command) {
        super(layerId);
        this.command = command;
    }

    @Override
    public boolean execute(ITimeline timeline) {
        String[] parts = command.split(" ", 2);

        return sendCommand(timeline, parts[0], parts.length > 1 ? parts[1] : "");
    }

    private boolean sendCommand(ITimeline timeline, String command, String parameters){
        AmcpLayer layer = timeline.getLayer(getLayerId());

        String[] parameterList = Arrays.stream(parameters.split(" ")).map(p -> timeline.getParameter(p)).toArray(s -> new String[s]);
        parameters = String.join(" ", parameterList);

        try {
            layer.sendCommand(command, parameters);
            return true;
        } catch (Exception e){
            System.err.println("Failed to execute command: " + e.getMessage());
            return false;
        }
    }

    public String getCommandName(){
        return "AmcpCommand";
    }

    @Override
    public String toString() {
        return getCommandName() + ": " + getLayerId() + " " + command;
    }
}

File: CViz/Timeline/Command/AmcpCommand.cs
using System;
using System.Linq;
using System.Text;
using log4net;
using StilSoft.CasparCG.AmcpClient.Commands;

namespace CViz.Timeline.Command
{
    class AmcpCommand : CommandBase
    {
        private static readonly ILog Log = LogManager.GetLogger(typeof(AmcpCommand));

        private readonly string _command;

        public AmcpCommand(int layerId, string command) : base(layerId)
        {
            _command = command;
        }

        public override bool Execute(ITimeline timeline)
        {
            string[] parts = _command.Split(new []{' '}, 2);

            return SendCommand(timeline, parts[0], parts.Length > 1 ? parts[1] : "");
        }

        private bool SendCommand(ITimeline timeline, string command, string parameters)
        {
            var translatedParameters = new StringBuilder();
            foreach (string param in parameters.Split(' ').Where(p => p.Length > 0))
            {
                string param2 = timeline.GetParameterValue(param, true);

                translatedParameters.Append("\"").Append(param2).Append("\" ");
            }

            try
            {
                new CustomCommand($"{command} {timeline.ChannelNumber}-{LayerId} {translatedParameters}").Execute(timeline.Client);
                return true;
            }
            catch (Exception e)
            {
                Log.ErrorFormat("Failed to execute command: {0}", e.Message);
                return false;
            }
        }

        protected virtual string CommandName => "AmcpCommand";

        public override string ToString()
        {
            return $"{CommandName}: {LayerId} {_command}";
        }
    }
}
